magento/magento2
# Only Update Indexer Mode When Required

### Description (*)
Updates the indexer mode change mass actions to only change modes when required i.e. when the current index mode is different to the one being applied. This prevents unnecessary trips to the database.

Also updates the messaging displayed to the admin to inform them how many indexers were updated and how many were skipped thanks to this change.

Finally, replaces the deprecated `addSuccess()` function with the non-deprecated `addSuccessMessage()` function for the message manager class when displaying the changed messages to the admin.

### Manual testing scenarios (*)
1. Log into the admin
2. Set any number of indexers to be `Update on Schedule`
3. Observe the success message to confirm the correct number of indexers were updated versus skipped, based on their original mode

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#36823: Only Update Indexer Mode When Required
